src/lib.rs

```diff
@@ -207,6 +209,23 @@ pub fn run(path: &PathBuf) -> Vec<Message> {
                             }
                         }
                     }
+
+                    if h.compressed_header && h.time_offset.is_some() {
+                        let time_offset = h.time_offset.unwrap();
+                        let date: u32 = if time_offset as u32 >= (last_timestamp & 0x1F) {
+                            (last_timestamp & 0xFFFFFFE0) + time_offset as u32
+                        } else {
+                            (last_timestamp & 0xFFFFFFE0) + time_offset as u32 + 0x20
+                        };
+                        unsafe {
+                            std::ptr::write(
+                                &mut datafield_buffer[valid_fields],
+                                DataField::new(253, Value::Time(date + PSEUDO_EPOCH)),
```

Comment:
Is there a danger that we'll somehow end up with two timestamps for this data record? Or does the spec say that (I'm a bit rusty) in compressed header cases a message doesn't ever contain its own timestamp?

Reply:
It's impossible due to the spec: the purpose of the compressed header is only to replace the timestamp field of the message by an offset stored in the header